### PR TITLE
Extend `sql-db/reactive-vanilla` with `mssql` datasource

### PR DESCRIPTION
**Pull request by jsmrcka**
Implementation of test plan for QUARKUS-1408:
https://github.com/quarkus-qe/quarkus-test-plans/blob/main/QUARKUS-1408.md

First commit contains refactoring of the module: https://github.com/quarkus-qe/quarkus-test-suite/commit/06cb4fa188af0c751b789b963eba45104661d643.
Second commit contains the `mssql` extension itself: https://github.com/quarkus-qe/quarkus-test-suite/commit/a85f922a5aa861ae64d8479a83b0a82dd47fa0ef.

### CODE REVIEW

File: sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/model/Book1.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.pgclient.PgPool;
+import io.vertx.mutiny.sqlclient.Row;
+import io.vertx.mutiny.sqlclient.RowSet;
+import io.vertx.mutiny.sqlclient.Tuple;
+
+public class Book1 extends Book {
```
Comment (mjurc) [inline sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/model/Book1.java:12]: Don't particularly like `Book1 extends Book` - can we have some other class names?

Like maybe `SoftCoverBook` and `HardCoverBook`? It's just that `Book1` vs `Book2` vs `Book` are too similar.

File: sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/model/Book2.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.mysqlclient.MySQLPool;
+import io.vertx.mutiny.sqlclient.Row;
+import io.vertx.mutiny.sqlclient.RowSet;
+import io.vertx.sqlclient.PropertyKind;
+
+public class Book2 extends Book {
```
Comment (mjurc) [inline sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/model/Book2.java:12]: Same note about the class name applies here.

File: sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/resources/Book1Resource.java [hunk 3]
```diff
@@ -12,46 +10,47 @@
...
 import io.quarkus.qe.model.Book;
-import io.quarkus.qe.model.Record;
+import io.quarkus.qe.model.Book1;
 import io.quarkus.runtime.StartupEvent;
 import io.smallrye.mutiny.Uni;
 import io.vertx.mutiny.pgclient.PgPool;
 
 @Path("/book/postgresql")
-public class BookResource extends CommonResource {
+public class Book1Resource extends CommonResource {
```
Comment (mjurc) [inline sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/resources/Book1Resource.java:24]: Same note about class naming applies here.

File: sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/resources/Book2Resource.java [hunk 4]
```diff
@@ -13,46 +11,48 @@
...
+import io.quarkus.qe.model.Book;
+import io.quarkus.qe.model.Book2;
 import io.quarkus.runtime.StartupEvent;
 import io.smallrye.mutiny.Uni;
 import io.vertx.mutiny.mysqlclient.MySQLPool;
 
-@Path("/notebook/mysql")
-public class NoteBookResource extends CommonResource {
+@Path("/book/mysql")
+public class Book2Resource extends CommonResource {
```
Comment (mjurc) [inline sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/resources/Book2Resource.java:25]: Same note about class name applies here.

**Review by mjurc — CHANGES_REQUESTED** [review]

File: sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/model/Book3.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.List;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.mssqlclient.MSSQLPool;
+import io.vertx.mutiny.sqlclient.Row;
+import io.vertx.mutiny.sqlclient.RowSet;
+import io.vertx.sqlclient.PropertyKind;
+
+public class Book3 extends Book {
```
Comment (mjurc) [inline sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/model/Book3.java:12]: Same note about class naming applies here.

File: sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/resources/Book3Resource.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+import io.quarkus.qe.model.Book;
+import io.quarkus.qe.model.Book3;
+import io.quarkus.runtime.StartupEvent;
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.mssqlclient.MSSQLPool;
+import io.vertx.mutiny.sqlclient.Pool;
+
+@Path("/book/mssql")
+public class Book3Resource extends CommonResource {
```
Comment (mjurc) [inline sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/resources/Book3Resource.java:26]: Same note about class naming applies here.

**Review by mjurc — CHANGES_REQUESTED** [review]

File: sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/model/Book1.java (hunk 1 above)
Comment (jsmrcka) [reply sql-db/reactive-vanilla/src/main/java/io/quarkus/qe/model/Book1.java]: Makes sense. It was originally named more diversely.

**Review by mjurc — APPROVED** [review]

**Pull request merged**